### PR TITLE
Move utils module and remove unused functions

### PR DESCRIPTION
**Pull request by therve**
This moves the utils module out of generation in the top package, and remove a bunch of unused functions.

### CODE REVIEW

File: api/v1/datadog/client.go [hunk 1]
```diff
@@ -582,60 +581,6 @@ func detectContentType(body interface{}) string {
...
-	}
-	return expires
-}
-
-func strlen(s string) int {
-	return utf8.RuneCountInString(s)
-}
-
 // GenericOpenAPIError Provides access to the body, error and model on returned errors.
 type GenericOpenAPIError struct {
```
Comment (jirikuncar) [inline api/v1/datadog/client.go:585]: should we move `GenericOpenAPIError` to top-level module too?

Comment (therve) [reply api/v1/datadog/client.go:585]: We could yeah.

File: utils.go [hunk 2]
```diff
@@ -4,7 +4,7 @@
  * Copyright 2019-Present Datadog, Inc.
  */
 
-package datadog
+package client
```
Comment (jirikuncar) [inline utils.go:7]: Is this file generated?

File: tests/api/v2/datadog/api_logs_archives_test.go [hunk 3]
```diff
@@ -11,8 +11,10 @@ import (
 	"fmt"
 	"testing"
 
+	utils "github.com/DataDog/datadog-api-client-go"
```
Comment (jirikuncar) [inline tests/api/v2/datadog/api_logs_archives_test.go:14]:
```suggestion
	dd "github.com/DataDog/datadog-api-client-go"
```

File: utils.go (hunk 2 above)
Comment (therve) [reply utils.go:7]: No. I moved it out as there were only static bits.

**Pull request closed without merging**